app/logic/Mail/ActiveSync/ActiveSyncFolder.ts

```diff
@@ -174,6 +174,17 @@ export class ActiveSyncFolder extends Folder implements ActiveSyncPingable {
     });
     this.messages.addAll(newMsgs);
     this.account.addPingable(this);
+    if (this.specialFolder == SpecialFolder.Sent) {
+      for (let email of newMsgs) {
+        let customSentFolder = this.account.customSentFolders.shift();
```

Comment:
* Save the sent emails in an array, instead of the `customSentFolders`. The email contains the property `.folder`.
* When `newMsgs` come in, you need to match up the new mail that comes in, with the email we sent. Either based message ID, if possible, or based on subject.